src/dns-test.js

```diff
@@ -0,0 +1,317 @@
+
+const TELEMETRY_TYPE = "dnssec-interference-study";
+const TELEMETRY_OPTIONS = {
+    addClientId: true,
+    addEnvironment: true
+};
+
+const MAX_TXID = 65535;
+const MIN_TXID = 0;
+
+var udp_query_proto;
+var tcp_query_proto;
+var measurement_id;
+
+var udpResponses = {"A":      {"data": "", "transmission": 0},
+                    "RRSIG":  {"data": "", "transmission": 0},
+                    "DNSKEY": {"data": "", "transmission": 0},
+                    "SMIMEA": {"data": "", "transmission": 0},
+                    "HTTPS":  {"data": "", "transmission": 0},
+                    "NEWONE": {"data": "", "transmission": 0},
+                    "NEWTWO": {"data": "", "transmission": 0}};
+
+var tcpResponses = {"A":      {"data": "", "transmission": 0},
+                    "RRSIG":  {"data": "", "transmission": 0},
+                    "DNSKEY": {"data": "", "transmission": 0},
+                    "SMIMEA": {"data": "", "transmission": 0},
+                    "HTTPS":  {"data": "", "transmission": 0},
+                    "NEWONE": {"data": "", "transmission": 0},
+                    "NEWTWO": {"data": "", "transmission": 0}};
+
+/**
+ * Encode a DNS query to be sent over a UDP socket
+ */
+function encodeUDPQuery(domain, rrtype) {
+    const buf = DNS_PACKET.encode({
+        type: 'query',
+        // Generate a random transaction ID between 0 and 65535
+        id: Math.floor(Math.random() * (MAX_TXID - MIN_TXID + 1)) + MIN_TXID,
+        flags: DNS_PACKET.RECURSION_DESIRED,
+        questions: [{
+            type: rrtype,
+            name: domain
+        }],
+        additionals: [{
+            type: 'OPT',
+            name: '.',
+            udpPayloadSize: 4096
+        }]
+    });
+    udp_query_proto = buf.__proto__;
+    return buf
+}
+
+/**
+ * Encode a DNS query to be sent over a TCP socket
+ */
+function encodeTCPQuery(domain, rrtype) {
+    const buf = DNS_PACKET.streamEncode({
+        type: 'query',
+        // Generate a random transaction ID between 0 and 65535
+        id: Math.floor(Math.random() * (MAX_TXID - MIN_TXID + 1)) + MIN_TXID,
+        flags: DNS_PACKET.RECURSION_DESIRED,
+        questions: [{
+            type: rrtype,
+            name: domain
+        }]
+    });
+    tcp_query_proto = buf.__proto__;
+    return buf;
+}
+
+/**
+ * Send a DNS query over UDP, re-transmitting according to default 
+ * resolvconf behavior if we fail to receive a response.
+ *
+ * In short, we re-transmit at most RESOLVCONF_ATTEMPTS for each nameserver 
+ * we find. The timeout for each missing response is RESOLVCONF_TIMEOUT 
+ * (5000 ms).
+ */
+async function sendUDPQuery(domain, nameservers, rrtype) {
+    let buf = encodeUDPQuery(domain, rrtype);
+    for (let i = 0; i < nameservers.length; i++) {
+        let written = 0;
+        for (let j = 1; j <= RESOLVCONF_ATTEMPTS; j++) {
+            try {
+                let nameserver = nameservers[i];
+                udpResponses[rrtype]["transmission"] += 1
```

Comment:
(code style)
While `udpResponses[rrtype]["transmission"]` is a valid syntax, it is more idiomatic to access properties without `["` and `"]`. Use `udpResponses[rrtype].transmission` instead.

Reply:
No longer relevant as of 60c7b378870db5efd371d975c1c13a04e5c9b58d. I now just use dnsData["udp" + rrtype] and dnsData["tcp" + rrtype]. This may not be super clean, though.